language-design.md

```diff
@@ -17,7 +17,82 @@ For each quote, describe what it was about the quote that led you pick it.
+notably, the quote contradicts itself.  The quote first says that if you
+have trouble intuiting code then it's not because you're a n00b.  However,
+they go on to reference a studying that tested people who have never coded
+before.  This quote does not support their point that experienced coders
+struggle with syntax.
+
+We also disagree with the sentiment that programmers should employ syntax more
+intuitive to novices.  Sure, using natural language could look more
+understandable for someone who has never coded before.  However, for more
+experienced coders, using natural language could feel ambiguous and cumbersome.
+Frequent users of code would appreciate a concise set of keywords that they
+could build off of.  Also, we think it is more important to cater to experienced
+coders than the "n00bs" in the study because they were only novices for an hour.
+Typically, people spend more time as experienced coders who understand the
+syntax than as beginners trying to read off code like a natural language.  There
+is certainly a niche for languages that resemble natural communication, for
+example a stepping stone to teach beginners computer science or a domain-
+specific language not intended for people with prior CS experience.  However,
+once a student has learned how to code, he or she will grow tired of the
+wordiness needed for natural language.
+
+2. From the article by Yang and Rabkin:
+
+> "While those on the outside are still struggling to prove themselves, the
+> technically privileged have gone ahead to determine what the software that
+> runs our lives should look like."
+> [Yang and Rabkin, 2015].
+
+Both of us spent the summer working at tech companies whose engineers were
+overwhelmingly male, so this quote struck us as being particularly accurate.
+Sadly, it epitomizes the current state of the tech industry as a place
+bifurcated into groups of "insiders" and "outsiders", one in which ability and
+potential have little to do with which group one finds themselves in.
+
+A particularly interesting point of the article, which we believe is
+encapsulated in this quote, is the ambiguous direction of causality when it
+comes to language-based discrimination. Do so-called "Real Programmers" look
+down on Javascript or HTML users because of prejudice against those languages?
+Or have industry insiders cultivated the language-based stereotypes as a means
+to discriminate against those who tend to use the languages? We believe the
+causality flows both ways. Those on the inside have certain prejudices about
+languages, which inform their prejudices about those who use the languages.
+They then utilize the language prejudice to keep the outsiders out and
+themselves in. Breaking this cycle of privilege will require a dramatic
+cultural shift.
+
+3. From the abstract by Bloch:
+
+> "You can't please everyone, so aim to displease everyone equally" [Bloch,
+2006]
+
+We chose this quote because it is an undesirable, yet important part of
```

Comment:
I disagree with the quote, to an extent. Trying to displease everyone equally seems a little silly, especially when there are a whole bunch of people you weren't trying to please anyway. In particular for a DSL, you should be aiming at displeasing domain experts equally, and ignoring the effect on everyone else. Maybe I'm being a bit too pedantic with my interpretation, but there are still some people who you should displease less than others, and that group is your target audience.